### PR TITLE
Tag list

### PR DESCRIPTION
**Pull request by hbjorgo**
Added more properties to Tag DTO. Added unit tests. Added code to seed test data.

### CODE REVIEW

**Comment by janid1967** [comment]
As requested pr mail last week. Want this naming:
CalloffNumber -> CallOff
CommPkgNumber -> CommPkgNo
McPkgNumber -> McPkgNo
PurchaseOrderNumber -> PONo (or PurchaseOrderNo)

File: src/Equinor.Procosys.Preservation.Query/TagAggregate/TagDto.cs [hunk 1]
```diff
@@ -6,23 +6,54 @@ namespace Equinor.Procosys.Preservation.Query.TagAggregate
...
+            PurchaseOrderNumber = purchaseOrderNumber;
             TagNo = tagNo;
             Status = status;
             StepId = stepId;
-            Requirements = requirements;
+            TagFunctionCode = tagFunctionCode;
+            Requirements = requirements ?? throw new ArgumentNullException(nameof(requirements));
         }
+
+        public string AreaCode { get; set; }
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/TagAggregate/TagDto.cs]: remove setters ... all of them

File: src/Equinor.Procosys.Preservation.WebApi/Seeding/Seeder.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Equinor.Procosys.Preservation.WebApi.Seeding
+{
+    public class Seeder : IHostedService
+    {
+        private readonly IServiceScopeFactory _serviceProvider;
+
+        public Seeder(IServiceScopeFactory serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Seeding/Seeder.cs]: expr. body

File: src/Equinor.Procosys.Preservation.WebApi/Seeding/Seeder.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    var responsibles = await responsibleRepository.GetAllAsync();
+                    journeyRepository.AddJourneys(250, plantProvider.Plant, modes, responsibles);
+                    await unitOfWork.SaveChangesAsync(cancellationToken);
+
+                    requirementTypeRepository.AddRequirementTypes(250, plantProvider.Plant);
+                    await unitOfWork.SaveChangesAsync(cancellationToken);
+
+                    var steps = (await journeyRepository.GetAllAsync()).SelectMany(x => x.Steps).ToList();
+                    var requirementDefinitions = (await requirementTypeRepository.GetAllAsync()).SelectMany(x => x.RequirementDefinitions).ToList();
+                    for (int i = 0; i < 1; i++)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Seeding/Seeder.cs]: var

File: src/Equinor.Procosys.Preservation.WebApi/Seeding/Seeders.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        public static void AddTags(this ITagRepository tagRepository, int entryCount, string plant, List<Step> steps, List<RequirementDefinition> requirementDefinitions)
+        {
+            var rand = new Random();
+
+            for (var i = 0; i < entryCount; i++)
+            {
+                var requirements = new List<Requirement>();
+                for (int j = 0; j < 5; j++)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Seeding/Seeders.cs]: var

File: src/Equinor.Procosys.Preservation.WebApi/Seeding/SeedingPlantProvider.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using Equinor.Procosys.Preservation.Domain;
+
+namespace Equinor.Procosys.Preservation.WebApi.Seeding
+{
+    public class SeedingPlantProvider : IPlantProvider
+    {
+        public SeedingPlantProvider(string plant)
+        {
+            Plant = plant;
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Seeding/SeedingPlantProvider.cs]: expr.body

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/TagAggregate/TagDtoTests.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                "ProjectNo",
+                "PoNo",
+                new List<RequirementDto> { new RequirementDto(1, 2, true, 3) },
+                PreservationStatus.Active,
+                2,
+                "TagFunctionCode",
+                "TagNo");
+
+        [TestMethod]
+        public void Constructor_setsProperties()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/TagAggregate/TagDtoTests.cs]: SetsProperties

**Comment by hbjorgo** [comment]
> As requested pr mail last week. Want this naming:
> CalloffNumber -> CallOff
> CommPkgNumber -> CommPkgNo
> McPkgNumber -> McPkgNo
> PurchaseOrderNumber -> PONo (or PurchaseOrderNo)

Done.

**Pull request merged**